openassessment/assessment/test/test_staff.py

```diff
@@ -382,6 +391,53 @@ def test_database_create_error_handling(self, mock_create):
         )
 
+    def test_fetch_next_submission(self):
+        bob_sub, bob = self._create_student_and_submission("bob", "bob's answer")
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer")
+        submission = staff_api.get_submission_to_assess(tim['course_id'], tim['item_id'], tim['student_id'])
+        self.assertIsNotNone(submission)
+        self.assertEqual(bob_sub, submission)
+
+    def test_fetch_same_submission(self):
+        bob_sub, bob = self._create_student_and_submission("bob", "bob's answer")
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer")
+        tim_to_grade = staff_api.get_submission_to_assess(tim['course_id'], tim['item_id'], tim['student_id'])
+        self.assertEqual(bob_sub, tim_to_grade)
+        # Ensure that Bob doesn't pick up the submission that Tim is grading.
+        bob_to_grade = staff_api.get_submission_to_assess(tim['course_id'], tim['item_id'], bob['student_id'])
+        tim_to_grade = staff_api.get_submission_to_assess(tim['course_id'], tim['item_id'], tim['student_id'])
+        self.assertEqual(bob_sub, tim_to_grade)
+        self.assertEqual(tim_sub, bob_to_grade)
+
+    def test_fetch_submission_delayed(self):
+        bob_sub, bob = self._create_student_and_submission("bob", "bob's answer")
+        # Fetch the submission for Tim to grade
+        tim_to_grade = staff_api.get_submission_to_assess(bob['course_id'], bob['item_id'], "Tim")
+        self.assertEqual(bob_sub, tim_to_grade)
+
+        # Change the grading_started_at timestamp so that the 'lock' on the
+        # problem is released.
+        workflow = StaffWorkflow.objects.get(scorer_id="Tim")
+        timestamp = (now() - (workflow.TIME_LIMIT + timedelta(hours=1))).strftime("%Y-%m-%d %H:%M:%S")
+        workflow.grading_started_at = timestamp
+        workflow.save()
+
+        bob_to_grade = staff_api.get_submission_to_assess(bob['course_id'], bob['item_id'], bob['student_id'])
+        self.assertEqual(tim_to_grade, bob_to_grade)
+
+    def test_next_submission_error(self):
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer")
+        with mock.patch('openassessment.assessment.api.staff.submissions_api.get_submission') as patched_get_submission:
+            patched_get_submission.side_effect = sub_api.SubmissionNotFoundError('Failed')
+            with self.assertRaises(staff_api.StaffAssessmentInternalError):
+                submission = staff_api.get_submission_to_assess(tim['course_id'], tim['item_id'], tim['student_id'])
+
+    def test_no_available_submissions(self):
+        tim_sub, tim = self._create_student_and_submission("Tim", "Tim's answer")
+        # Use a non-existent course and non-existent item.
+        submission = staff_api.get_submission_to_assess('test_course_id', 'test_item_id', tim['student_id'])
+        self.assertIsNone(submission)
+
```

Comment:
If staff workflows can be cancelled, it would be good to add a test for the behavior in that case.